### DISCUSSION TITLE
Fetch user roles from Zitadel api

### DISCUSSION THREAD

**Discussion by lucasKuratani** [zitadel/zitadel#3993] [Q&A] [answered]
Hello, I'm new to Zitadel and I'm trying to fetch the user roles from the api.

I can fetch the user info from this endpoint: {domain}/oidc/v1/userinfo, with the access token as the authorization header.

I was wondering if it was possible to fetch the user roles with this very method, so I can give authorization to my protected api routes.

If that's not the way I should be doing this, I'd apreciate some advice.

Thanks in advance.

**Comment by fforootd** [comment, marked as answer]
Hi @lucasKuratani 

I am happy to help.

ZITADEL currently support multiple ways of reading the user roles and authorizations.

1. [You can request specified roles as scopes](https://docs.zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes) e.g `urn:zitadel:iam:org:project:role:user` (The role will he handled in the same way as in the next step)
2. You can assert the roles generally through the projects settings "Assert Roles on Authentication". In this case the roles are written to the tokens as well as the [userinfo](https://docs.zitadel.com/docs/apis/openidoauth/endpoints#userinfo_endpoint) (used for apps) and [introspect endpoint ](https://docs.zitadel.com/docs/apis/openidoauth/endpoints#introspection_endpoint) (used from headless apis who only consume the access_token)
3. You can also access the users authorizations through the [management api](https://editor.swagger.io/?url=https://zitadel.cloud/openapi/v2/swagger/management.swagger.json) with a service user which also could change authorizations. You can find a [go example here ](https://docs.zitadel.com/docs/quickstarts/call-zitadel-api/go)

May I ask what technology you are using. Maybe I can shine some more details to it.
